### PR TITLE
Issue 88 - Allow user to change payment method

### PR DESCRIPTION
**Pull request by ahmed-shehata**
Issue #88 - Allow user to change payment method

### CODE REVIEW

File: AtlasUI/AtlasUI/CheckoutSummaryRow.swift [hunk 1]
```diff
@@ -42,7 +41,7 @@ internal final class CheckoutSummaryRow: UIView, UIGestureRecognizerDelegate {
         detailTextLabel.topAnchor.constraintEqualToAnchor(self.topAnchor).active = true
         detailTextLabel.bottomAnchor.constraintEqualToAnchor(self.bottomAnchor).active = true
 
-        if onTap != nil {
+        if tapAction != nil {
```
Comment (bonkey) [inline AtlasUI/AtlasUI/CheckoutSummaryRow.swift]: I'd consider something like this:

```
var tapAction: (() -> Void)? {
  didSet {
    if tapAction != nil {
             arrowImageView.image = UIImage(named: "arrow", bundledWith: CheckoutSummaryRow.self)
             arrowImageView.translatesAutoresizingMaskIntoConstraints = false
//...
    } else {
      arrowImageView.removeFromSuperview()
    }
  }
```

However not always the best to have some implicit behaviour.

File: AtlasUI/AtlasUI/PaymentSelectionViewController.swift [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+            return true
+    }
+
+    func webView(webView: UIWebView, didFailLoadWithError error: NSError?) {
+        dismissViewController(.failure(), animated: true)
+    }
+
+    private func dismissViewController(result: Result<String>, animated: Bool = true) {
+        navigationController?.popViewControllerAnimated(animated)
+        self.paymentCompletion?(result)
```
Comment (bonkey) [inline AtlasUI/AtlasUI/PaymentSelectionViewController.swift]: Either self in both lines or in none.

File: AtlasUI/AtlasUI/PaymentSelectionViewController.swift [hunk 3]
```diff
@@ -0,0 +1,67 @@
+//
+//  Copyright © 2016 Zalando SE. All rights reserved.
+//
+
+import UIKit
+
+enum Result<T> {
```
Comment (bonkey) [inline AtlasUI/AtlasUI/PaymentSelectionViewController.swift]: Why don't you use `AtlasResult`?

**Comment by bonkey** [comment]
👍

File: AtlasUI/AtlasUI/PaymentSelectionViewController.swift (hunk 3 above)
Comment (ahmed-shehata) [inline AtlasUI/AtlasUI/PaymentSelectionViewController.swift]: Different scope (AtlasSDK)

**Comment by ahmed-shehata** [comment]
👍

**Comment by bonkey** [comment]
:+1:

File: AtlasUI/AtlasUI/PaymentSelectionViewController.swift (hunk 3 above)
Comment (bonkey) [inline AtlasUI/AtlasUI/PaymentSelectionViewController.swift]: Isn't it public?

**Pull request merged**